01-messaging.md

```diff
@@ -225,11 +225,11 @@ Various fundamental types are referred to in the message specifications:
 * `u64`: an 8 byte unsigned integer
 
 Inside TLV records which contain a single value, leading zeros in
-integers can be omitted:
+truncated integers MUST be omitted:
```

Comment:
Truncated integers are not limited to TLV records with single values, and in fact we already use them in onion routing TLVs with multiple values.

Since we're making changes here, I would be in favor of also clarifying where truncated integers can be used.

```md
For the final value in TLV records, truncated integers may be used.  Leading zeros in
truncated integers MUST be omitted:
```